mesa/agent.py

```diff
@@ -17,7 +19,7 @@ def __init__(self, unique_id: int, model: Model) -> None:
         """ Create a new agent. """
         self.unique_id = unique_id
         self.model = model
-        self.pos = None
+        self.pos: Optional[Union[Coordinate, FloatCoordinate, int]] = None
```

Comment:
I think it would be cleaner to abstract out `Union[Coordinate, FloatCoordinate, int]`, so that when new coordinate type is added, agent.py will stay the same.

Reply:
Do you mean putting something like `Position = Union[Coordinate, FloatCoordinate, int]` into `space.py` and importing this in agent.py`?

Reply:
Yes.

Reply:
Want to know what Agents.jl does for this? They define separate agent struct for each space type, e.g. `ContinuousAgent` https://github.com/JuliaDynamics/Agents.jl/blob/0c660872f983b5a981671bc8f1b5c901c8799a93/src/core/agents.jl#L100

Reply:
This has never been a problem in Python because of duck typing and that methods can have any type, but once we go to explicit typing, we get mess like in Agents.jl.